GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java

```diff
@@ -256,7 +262,8 @@ public void setMain(GVRMain gvrMain, String dataFileName) {
             if (mConfigurationManager.isDockListenerRequired()) {
                 startDockEventReceiver();
             } else {
-                handleOnDock();
+                //android.util.Log.w(TAG, "setMain calls handleOnDock decided by isDockListenerRequired");
+                //handleOnDock();
```

Comment:
This is the earliest possible place where handleOnDock can be invoked and have the backend correctly initialized. Let me know in an email why this didn't work for you, thanks.